src/providers/node.rs

```diff
@@ -36,7 +36,12 @@ impl Provider for NodeProvider {
 
     fn setup(&self, app: &App, env: &Environment) -> Result<Option<SetupPhase>> {
         let packages = NodeProvider::get_nix_packages(app, env)?;
-        Ok(Some(SetupPhase::new(packages)))
+        let mut setup_phase = SetupPhase::new(packages);
+        if NodeProvider::uses_canvas(app) {
```

Comment:
Idea: for the future (I'll implement) a method to check if a `package.json` has a dependency. Should check only `dependencies`.